lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp

```diff
@@ -348,13 +340,16 @@ bool fastPathAvailable(const SharedMemoryObject &smemObj,
 // @param numOfElems number of elements accessed by thread per repetition
 // @param reps number of instructions repretition to fully cover dot operand
 // @param cSwizzleOffset
+// TODO (zhanglx): We should never call this function because we always have
```

Comment:
This make sense. I think you can entirely remove this function and add a check in `fastPathAvailable`.

Reply:
I removed this function. fastPathAvailable looks good to me. What check are you referring to?

Reply:
I was talking about additional condition checking that operand is k-major.

In some experiments I (and probably others as well) disable swizzling. In this case I think it is better to simply fall-back to normal path.